source/gui/settingsDialogs.py

```diff
@@ -3009,92 +3015,158 @@ def safeTerminateProviders(
 
 	def onSave(self):
 		for panel in self.providerPanelInstances:
-			panel.onSave()
+			try:
+				panel.onSave()
+			except:
+				log.debug(f"Error saving providerPanel: {panel.__class__!r}", exc_info=True)
 		self.initialProviders = list(vision.handler.providers)
 
 
-class VisionProviderSubPanel(DriverSettingsMixin, SettingsPanel):
-
+class VisionProviderSubPanel_Settings(
+		DriverSettingsMixin,
+		SettingsPanel
+):
+	cachePropertiesByDefault = False
 	def __init__(
 			self,
 			parent: wx.Window,
 			*,  # Make next argument keyword only
-			providerCallable: Callable[[], vision.providerBase.VisionEnhancementProvider]
+			settingsCallable: Callable[[], vision.providerBase.VisionEnhancementProviderStaticSettings]
 	):
 		"""
 		@param providerCallable: A callable that returns an instance to a VisionEnhancementProvider.
 			This will usually be a weakref, but could be any callable taking no arguments.
 		"""
-		self._providerCallable = providerCallable
+		self._settingsCallable = settingsCallable
 		super().__init__(parent=parent)
 
 	@property
 	def driver(self):
-		return self._providerCallable()
+		return self.getSettings()
+
+	def getSettings(self):
+		settings = self._settingsCallable()
+		log.debug(f"getting settings: {settings.__class__!r}")
+		return settings
 
 	def makeSettings(self, settingsSizer):
 		# Construct vision enhancement provider settings
 		self.updateDriverSettings()
 
 
+class VisionProviderSubPanel_Wrapper(
+		SettingsPanel
+):
+
+	def __init__(
+			self,
+			parent: wx.Window,
+			*,  # Make next argument keyword only
+			# todo: make these part of a class:
+			providerType: Type[vision.VisionEnhancementProvider],
+			getProvider: Callable[
+				[],
+				Optional[vision.VisionEnhancementProvider]
+			],# mostly used to see if the provider is initialised or not.
+			initProvider: Callable[
+				[],
+				bool
+			],
+			terminateProvider: Callable[
+				[],
+				None
+			]
+	):
+		"""
+		@param getProvider: A callable that returns an instance to a VisionEnhancementProvider.
+			This will usually be a weakref, but could be any callable taking no arguments.
+		"""
+		self._providerType = providerType
+		self._getProvider = getProvider
+		self._initProvider = initProvider
+		self._terminateProvider = terminateProvider
+		self._runtimeSettings: Optional[VisionProviderSubPanel_Settings] = None
+		self._runtimeSettingsSizer = wx.BoxSizer(orient=wx.VERTICAL)
+		super().__init__(parent=parent)
+
+	def makeSettings(self, settingsSizer):
+		checkBox = wx.CheckBox(self, label=_("Enable"))
+		settingsSizer.Add(checkBox)
+		settingsSizer.Add(self._runtimeSettingsSizer, flag=wx.EXPAND, proportion=1.0)
+		self._checkBox: wx.CheckBox = checkBox
+		if self._getProvider():
+			checkBox.SetValue(True)
+		if self._createRuntimeSettings():
+			checkBox.Bind(wx.EVT_CHECKBOX, self._enableToggle)
+		else:
+			checkBox.Bind(wx.EVT_CHECKBOX, self._nonEnableableGUI)
+
+	def _createRuntimeSettings(self):

```

Comment:
This name should be changed, it's a left over from an earlier idea. 

```suggestion
	def _createSettings(self):
```